addons/account/models/account_journal.py

```diff
@@ -323,6 +324,44 @@ def _compute_suspense_account_id(self):
             else:
                 journal.suspense_account_id = False
 
+    def _inverse_type(self):
+        for record in self:
+            # Create an alias for purchase/sales journals
+            if record.type in ('purchase', 'sale'):
+                for alias_name in [
+                    record.alias_name,
+                    record.name,
+                    record.code,
+                    record.type,
+                ]:
+                    if alias_name:
+                        try:
+                            # ensure the default value can be encoded and used
+                            remove_accents(alias_name).encode('ascii')
+                        except UnicodeEncodeError:
+                            continue
+                        break
+                if record.company_id != self.env.ref('base.main_company'):
+                    alias_name = f"{alias_name}-{record.company_id.name}"
+
+                alias_values = {
+                    'alias_defaults': {
+                        'move_type': record.type == 'purchase' and 'in_invoice' or 'out_invoice',
+                        'company_id': record.company_id.id,
+                        'journal_id': record.id,
+                    },
+                    'alias_parent_thread_id': record.id,
+                    'alias_name': alias_name,
+                }
+                if record.alias_id:
+                    record.alias_id.sudo().write(alias_values)
+                else:
+                    alias_values['alias_model_id'] = self.env['ir.model']._get('account.move').id
+                    alias_values['alias_parent_model_id'] = self.env['ir.model']._get('account.journal').id
+                    record.alias_id = self.env['mail.alias'].sudo().create(alias_values)
+            elif record.alias_id:
+                record.alias_id.sudo().unlink()
```

Comment:
I tried to use `mail.alias.mixin`, but there is just way too much that doesn't work out of the box
* `alias_id` is required, but we only want it for sales/purchases journals
* it is using a `write`/`create` mechanism, forcing to use a mix of values to be written and existing values
* not clear to know when to update what, this kept it simple: never update the alias except when changing the type